### PR TITLE
Allow function call dependency declarations

### LINKED ISSUE #18: Function parsing issue
The log says that there is an `extraneous input '(' expecting {'}'`.

Looks like it is caused because I am calling a function inside `implementation`.

```kotlin
plugins {
    alias(libs.plugins.javiersc.hubdle)
    id("com.squareup.sort-dependencies") version "0.1"
}

hubdle {
    config {
        explicitApi()
        publishing()
    }
    kotlin {
        multiplatform {
            common {
                test {
                    dependencies {
						// sort these dependencies
                        implementation(projects.mokokiTest)
                        implementation(javierscKotlinStdlib())
                    }
                }
            }

            android()

            darwin {
                enableAll()
            }

            jvm()
            jvmAndAndroid()

            mingw {
                enableAll()
            }
        }
    }
}
```

I totally understand that supporting each use case can be out of the scope of this plugin and I am not sure if it is a "legal" use case to use functions, so feel free to close it if you think that this shouldn't be fixed.

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #18

### CODE REVIEW

**Comment by ZacSweers** [comment]
Draft because, for the life of me, I don't understand why this works

**Comment by ZacSweers** [comment]
It seems this does work

File: grammar/src/test/groovy/com/squareup/grammar/GradleGroovyScriptSpec.groovy [hunk 1]
```diff
@@ -348,14 +356,14 @@ final class GradleGroovyScriptSpec extends Specification {
...
-          
+
           /**
            * @return The JDK, for use by Proguard.
            */
           List<File> javaRuntime() {
             Jvm jvm = Jvm.current()
             FilenameFilter filter = { _, fileName -> fileName.endsWith(".jar") || fileName.endsWith(".jmod") }
-          
+
```
Comment (autonomousapps) [inline grammar/src/test/groovy/com/squareup/grammar/GradleGroovyScriptSpec.groovy:366]: not to be that guy, but are all these random whitespace changes necessary? :-P

**Review by autonomousapps — APPROVED** [review]
LGTM.

**Pull request merged**